xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -205,8 +216,9 @@ public DownstreamTlsContext getDownstreamTlsContext(Channel channel) {
    * @param remoteInetAddr source address of the inbound connection
    */
   private DownstreamTlsContext getDownstreamTlsContext(
-      InetSocketAddress localInetAddr, InetSocketAddress remoteInetAddr) {
-    List<FilterChain> filterChains = curListener.getFilterChains();
+      InetSocketAddress localInetAddr, InetSocketAddress remoteInetAddr,
+      EnvoyServerProtoData.Listener listener) {
```

Comment:
With `listener` being passed in you can make this `getDownstreamTlsContext` static after making `filterOnSourceType` and `filterOnIpAddress static too (which they should already have been).

Reply:
yea we can do